packages/webapp/webapp_server.js

```diff
@@ -812,22 +804,82 @@ var runWebAppServer = function () {
+        onListeningCallbacks = null;
+        _.each(callbacks, function (x) { x(); });
+
+      }, function (e) {
+        console.error("Error listening:", e);
+        console.error(e && e.stack);
+      }));
+    };
+
+    localPort = isNaN(Number(localPort)) ? localPort : Number(localPort);
 
-    }, function (e) {
-      console.error("Error listening:", e);
-      console.error(e && e.stack);
-    }));
+    if (typeof localPort == "string") {
+      var socketPath = localPort;
+      var listenOptions = { path: socketPath };
 
+      // Test to see if the socket path is Windows Server-style named pipe with the
+      // format: \\.\pipe\{randomstring} or \\{servername}\pipe\{randomstring}
+      // Otherwise, treat it as a UNIX domain socket path pointing to a socket file
+      if ( /\\\\?.+\\pipe\\?.+/.test(socketPath)) {
+        startHttpServer(listenOptions);
+      } else {
+        if (fs.existsSync(socketPath)) {
+          if (fs.statSync(socketPath).isSocket()) {
+            var clientSocket = new net.Socket();
+            clientSocket.on('error', Meteor.bindEnvironment(function(e) {
+              //If there is an existing socket file that we cannot connect to,
+              //it must be stale as it is not associated with a running server
+              if (e.code == 'ECONNREFUSED') {
+                console.log("Deleting stale socket file");
+                fs.unlinkSync(socketPath);
+                startHttpServer(listenOptions);
+              }
+            }));
+            clientSocket.connect({ path: socketPath }, function() {
+              //If there is a running server listening on that socket file,
+              //we have to leave the file alone to avoid preventing other clients
+              //from connecting to it.
+              console.error("Another server is already listening on socket: " + socketPath + ", exiting.");
+              process.exit();
+            });
+          } else {
+            //There is an existing file at socketPath that is not a socket,
+            //so we will leave it alone and exit
+            console.error("Cannot listen on socket: " + socketPath + ", exiting.");
+            process.exit();
+          }
+        } else {
+          startHttpServer(listenOptions);
+        }
+        //Clean up the socket file when we've finished to avoid leaving a stale one
+        //That situation should only be able to happen if the running node process is
+        //killed via signal 9 (SIGKILL).
+        process.on('exit', Meteor.bindEnvironment(function(e) {
+          fs.unlinkSync(socketPath);
+        }));
+        process.on('SIGINT', Meteor.bindEnvironment(function(e) {
+          process.exit();
+        }));
+      }
+    } else if (typeof localPort == "number") {
+      var listenOptions = { port: localPort, host: host };
+      startHttpServer(localPort);
+    } else {
+      console.error("Invalid PORT specified");
+      process.exit();
```

Comment:
I don't think we should be calling `process.exit` anywhere in this code.  That will be handled by the caller to `main()` [here in `boot.js`](https://github.com/meteor/meteor/blob/devel/tools/static-assets/server/boot.js#L353).  I think you can just return `ERROR` or something (though you should leave the `console.error`.